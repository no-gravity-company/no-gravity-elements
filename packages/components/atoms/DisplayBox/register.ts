import register from 'preact-custom-element';
import DisplayBox from './DisplayBox';

const alreadyDefined = (tagName: string) =>
    customElements.get(tagName) !== undefined;
if (!alreadyDefined('nge-display-box')) {
    register(DisplayBox, 'nge-display-box', ['label'], { shadow: true });
}
