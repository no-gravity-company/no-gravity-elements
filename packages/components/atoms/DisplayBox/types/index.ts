export interface DisplayBoxProps {
    label: string;
    value: number;
}
