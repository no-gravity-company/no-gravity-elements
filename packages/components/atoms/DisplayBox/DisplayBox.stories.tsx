import { h } from 'preact';
import { Story } from '@storybook/preact';
import { html } from 'lit-html';

import { DisplayBoxProps } from '@atoms/DisplayBox/types';

import '@atoms/DisplayBox/lib/index';

export default {
  title: 'DisplayBox',
  component: 'nge-display-box',
  parameters: {
    // Specify that the component is a Web Component
    webComponents: {
      // Set the tag name of the Web Component
      tagName: 'nge-display-box',
    },
  },
};

export const Default: Story<DisplayBoxProps> = (args): any => {
  return html` <nge-display-box /> `;
};
