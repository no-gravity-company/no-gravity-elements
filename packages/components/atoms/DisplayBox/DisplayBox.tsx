import { Fragment, FunctionComponent, h } from 'preact';
import { DisplayBoxProps } from './types';

const DisplayBox: FunctionComponent<DisplayBoxProps> = ({
    label = '',
    value = 0,
}: DisplayBoxProps) => {
    const handleClick = () => {
        console.log('clicked');
    };

    return (
        <div>
            <span>DISPLAY BOX</span>
            <span>{label}</span>
            <span>value</span>
            <span>{value}</span>
            <button onClick={handleClick}></button>
        </div>
    );
};

export default DisplayBox;
