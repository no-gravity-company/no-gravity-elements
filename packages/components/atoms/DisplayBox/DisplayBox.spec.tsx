import { shallow } from 'enzyme';
import { h } from 'preact';
import DisplayBox from './DisplayBox';

describe('display-box web component snapshot test', () => {
    it.each`
        label    | value
        ${'foo'} | ${1}
        ${'bar'} | ${5}
    `('should match the snapshot', ({ label, value }: any) => {
        const wrapper = shallow(<DisplayBox label={label} value={value} />);
        expect(wrapper).toMatchSnapshot();
    });
});
