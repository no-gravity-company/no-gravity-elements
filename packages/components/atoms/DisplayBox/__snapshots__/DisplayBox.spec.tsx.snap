// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display-box web component snapshot test should match the snapshot 1`] = `
<div>
  <span>
    DISPLAY BOX
  </span>
  <span>
    foo
  </span>
  <span>
    value
  </span>
  <span>
    1
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`display-box web component snapshot test should match the snapshot 2`] = `
<div>
  <span>
    DISPLAY BOX
  </span>
  <span>
    bar
  </span>
  <span>
    value
  </span>
  <span>
    5
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;
